crates/router/src/connector/volt/transformers.rs

```diff
@@ -309,37 +314,118 @@ pub enum VoltPaymentStatus {
+                                .merchant_internal_reference
+                                .or(Some(payment_response.id)),
+                            incremental_authorization_allowed: None,
+                        })
+                    },
+                    ..item.data
+                })
+            }
+            VoltPaymentsResponseData::WebhookResponse(webhook_response) => {
+                let detailed_status = webhook_response.detailed_status.clone();
+                let status = enums::AttemptStatus::from(webhook_response.status);
+                Ok(Self {
+                    status,
+                    response: if is_payment_failure(status) {
+                        Err(types::ErrorResponse {
+                            code: detailed_status
+                                .clone()
+                                .map(|volt_status| volt_status.to_string())
+                                .unwrap_or_else(|| consts::NO_ERROR_CODE.to_owned()),
+                            message: detailed_status
+                                .clone()
+                                .map(|volt_status| volt_status.to_string())
+                                .unwrap_or_else(|| consts::NO_ERROR_MESSAGE.to_owned()),
+                            reason: detailed_status
+                                .clone()
+                                .map(|volt_status| volt_status.to_string()),
+                            status_code: item.http_code,
+                            attempt_status: None,
+                            connector_transaction_id: Some(webhook_response.payment.clone()),
+                        })
+                    } else {
+                        Ok(types::PaymentsResponseData::TransactionResponse {
+                            resource_id: types::ResponseId::ConnectorTransactionId(
+                                webhook_response.payment.clone(),
+                            ),
+                            redirection_data: None,
+                            mandate_reference: None,
+                            connector_metadata: None,
+                            network_txn_id: None,
+                            connector_response_reference_id: webhook_response
+                                .merchant_internal_reference
+                                .or(Some(webhook_response.payment)),
+                            incremental_authorization_allowed: None,
+                        })
+                    },
+                    ..item.data
+                })
+            }
+        }
+    }
+}
+
+impl From<VoltWebhookStatus> for enums::AttemptStatus {
+    fn from(status: VoltWebhookStatus) -> Self {
+        match status {
+            VoltWebhookStatus::Completed | VoltWebhookStatus::Received => Self::Charged,
+            VoltWebhookStatus::Failed | VoltWebhookStatus::NotReceived => Self::Failure,
+            VoltWebhookStatus::Pending => Self::AuthenticationPending,
```

Comment:
Could you please add the definition of `VoltWebhookStatus::Pending` in comments, then we ll know the reason why this is mapped to Authentication Pending, to avoid confusion in future